app/src/main/java/com/eveningoutpost/dexdrip/cgm/connectfollow/ConnectClient.java

```diff
@@ -0,0 +1,346 @@
+package com.eveningoutpost.dexdrip.cgm.connectfollow;
+
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.Getter;
+import okhttp3.*;
+
+import com.eveningoutpost.dexdrip.cgm.connectfollow.messages.ConnectData;
+import com.eveningoutpost.dexdrip.cgm.connectfollow.messages.ConnectDataResult;
+import com.google.gson.*;
+
+
+
+// TODO list
+// 1) automatic token handling
+//    - request token
+//    - refresh token
+//    - remove fetch data
+// 2) add other CareLink APIs
+//    - get patient data
+//    - get device info (last update???)
+//    - generate report
+//    - others?!
+// 3) optimize getLast24hours
+//    - use device info
+// 4) HTML helper class for data extraction from responses
+// 6) other events, markers, notes
+
+public class ConnectClient {
+    protected String carelinkUsername;
+    protected String carelinkPassword;
+    protected Country carelinkCountry;
+    protected static final String CARELINK_CONNECT_SERVER_EU = "carelink.minimed.eu";
+    protected static final String CARELINK_CONNET_SERVER_US = "carelink.minimed.com";
+    protected static final String CARELINK_LANGUAGE_EN = "en";
+    protected static final String CARELINK_LOCALE_EN = "en";
+    protected static final String CARELINK_AUTH_TOKEN_COOKIE_NAME = "auth_tmp_token";
+    protected static final String CARELINK_TOKEN_VALIDTO_COOKIE_NAME = "c_token_valid_to";
+    protected static final int AUTH_EXPIRE_DEADLINE_MINUTES = 1;
+
+    protected OkHttpClient httpClient = null;
+
+    private boolean lastLoginSuccess;
+    public boolean getLastLoginSuccess(){
+        return lastLoginSuccess;
+    }
+    private boolean lastDataSuccess;
+    public boolean getLastDataSuccess(){
+        return lastDataSuccess;
+    }
+    private String lastErrorMessage;
+    public String getLastErrorMessage(){
+        return lastErrorMessage;
+    }
+    private String lastStackTraceString;
+    public String getLastStackTraceString(){
+        return lastStackTraceString;
+    }
+
```

Comment:
Please apply code formatting to all of your code files and avoid unnecessary blank lines.